test/Conversion/HandshakeToHW/test_simple_loop.mlir

```diff
@@ -0,0 +1,73 @@
+// CHECK-SAME:                                                       %[[VAL_0:.*]]: !esi.channel<index>,
+// CHECK-SAME:                                                       %[[VAL_1:.*]]: !esi.channel<index>) -> (out0: !esi.channel<index>)
+// CHECK:         hw.module.extern @handshake_cond_br_in_ui1_2ins_2outs_ctrl(%[[VAL_2:.*]]: !esi.channel<i1>, %[[VAL_3:.*]]: !esi.channel<none>) -> (outTrue: !esi.channel<none>, outFalse: !esi.channel<none>)
+// CHECK:         hw.module.extern @handshake_sink_in_ui64(%[[VAL_0]]: !esi.channel<index>)
+// CHECK:         hw.module.extern @handshake_cond_br_in_ui1_ui64_out_ui64_ui64(%[[VAL_2]]: !esi.channel<i1>, %[[VAL_3]]: !esi.channel<index>) -> (outTrue: !esi.channel<index>, outFalse: !esi.channel<index>)
+// CHECK:         hw.module.extern @handshake_fork_in_ui1_out_ui1_ui1_ui1_ui1_ui1(%[[VAL_0]]: !esi.channel<i1>, %[[VAL_4:.*]]: i1, %[[VAL_5:.*]]: i1) -> (out0: !esi.channel<i1>, out1: !esi.channel<i1>, out2: !esi.channel<i1>, out3: !esi.channel<i1>, out4: !esi.channel<i1>)
+// CHECK:         hw.module.extern @arith_cmpi_in_ui64_ui64_out_ui1_slt(%[[VAL_0]]: !esi.channel<index>, %[[VAL_1]]: !esi.channel<index>) -> (out0: !esi.channel<i1>)
+// CHECK:         hw.module.extern @handshake_fork_in_ui64_out_ui64_ui64(%[[VAL_0]]: !esi.channel<index>, %[[VAL_4]]: i1, %[[VAL_5]]: i1) -> (out0: !esi.channel<index>, out1: !esi.channel<index>)
+// CHECK:         hw.module.extern @handshake_mux_in_ui1_ui64_ui64_out_ui64(%[[VAL_6:.*]]: !esi.channel<i1>, %[[VAL_0]]: !esi.channel<index>, %[[VAL_1]]: !esi.channel<index>) -> (out0: !esi.channel<index>)
+// CHECK:         hw.module.extern @handshake_mux_in_ui1_3ins_1outs_ctrl(%[[VAL_6]]: !esi.channel<i1>, %[[VAL_0]]: !esi.channel<none>, %[[VAL_1]]: !esi.channel<none>) -> (out0: !esi.channel<none>)
+// CHECK:         hw.module.extern @handshake_fork_in_ui1_out_ui1_ui1_ui1_ui1(%[[VAL_0]]: !esi.channel<i1>, %[[VAL_4]]: i1, %[[VAL_5]]: i1) -> (out0: !esi.channel<i1>, out1: !esi.channel<i1>, out2: !esi.channel<i1>, out3: !esi.channel<i1>)
+// CHECK:         hw.module.extern @handshake_buffer_in_ui1_out_ui1_1slots_seq(%[[VAL_0]]: !esi.channel<i1>, %[[VAL_4]]: i1, %[[VAL_5]]: i1) -> (out0: !esi.channel<i1>)
+// CHECK:         hw.module.extern @handshake_constant_c42_out_ui64(%[[VAL_7:.*]]: !esi.channel<none>) -> (out0: !esi.channel<index>)
+// CHECK:         hw.module.extern @handshake_constant_c1_out_ui64(%[[VAL_7]]: !esi.channel<none>) -> (out0: !esi.channel<index>)
+// CHECK:         hw.module.extern @handshake_fork_1ins_4outs_ctrl(%[[VAL_0]]: !esi.channel<none>, %[[VAL_4]]: i1, %[[VAL_5]]: i1) -> (out0: !esi.channel<none>, out1: !esi.channel<none>, out2: !esi.channel<none>, out3: !esi.channel<none>)
+
+// CHECK-LABEL:   hw.module @main(
+// CHECK-SAME:                    %[[VAL_0:.*]]: !esi.channel<none>,
+// CHECK-SAME:                    %[[VAL_4:.*]]: i1,
+// CHECK-SAME:                    %[[VAL_5:.*]]: i1) -> (out0: !esi.channel<index>, outCtrl: !esi.channel<none>) {
```

Comment:
so basically you've made clock and reset explicit here?

Reply:
Correct.